web/src/components/survey/question-typesII.ts

```diff
@@ -280,24 +280,35 @@ function initInfoText(Survey) {
         outer.appendChild(body);
       }
 
-      if (question.isRequired && !question.value) {
-        const acceptRow = document.createElement("div");
-        acceptRow.className = "row accept-row";
-        const cell = document.createElement("div");
-        cell.className = "col-sm-12";
-        const acceptBtn = document.createElement("button");
-        acceptBtn.className = "btn btn-primary";
-        const acceptLbl = document.createElement("span");
-        acceptLbl.appendChild(document.createTextNode("Continue"));
-        acceptBtn.appendChild(acceptLbl);
-        acceptBtn.addEventListener("click", () => {
-          question.value = 1;
-          acceptRow.style.display = "none";
-        });
-        cell.appendChild(acceptBtn);
-        acceptRow.appendChild(cell);
-        outer.appendChild(acceptRow);
+      const showContinueButton = () => {
+        if (question.isRequired && !question.value) {
+          const acceptRow = document.createElement("div");
+          acceptRow.className = "row accept-row";
+          const cell = document.createElement("div");
+          cell.className = "col-sm-12";
+          const acceptBtn = document.createElement("button");
+          acceptBtn.className = "btn btn-primary";
+          const acceptLbl = document.createElement("span");
+          acceptLbl.appendChild(document.createTextNode("Continue"));
+          acceptBtn.appendChild(acceptLbl);
+          acceptBtn.addEventListener("click", () => {
+            question.value = 1;
+            acceptRow.style.display = "none";
+          });
+          cell.appendChild(acceptBtn);
+          acceptRow.appendChild(cell);
+          outer.appendChild(acceptRow);
+        }
       }
+      showContinueButton();
+
+      question.registerFunctionOnPropertyValueChanged("isRequired", () => {
+        if (!question.isRequired) {
+          el.getElementsByClassName("row accept-row").forEach(e => e.remove());
+        }
+        else
```

Comment:
I'd recommend using the full brackets for the else, easier to debug if code is transpiled.. also the if used brackets.

Reply:
Got it. It will be changed in my next PR.